models/_sector/nft/trades/chains/ethereum/platforms/sudoswap_v2_ethereum_base_trades.sql

```diff
@@ -0,0 +1,175 @@
+            ,sp.call_tx_hash
+            ,sp.call_block_time
+            ,sp.call_block_number
+        FROM (
+            SELECT 
+                *
+                , case when cardinality(call_trace_address) = 0 then array[0] else call_trace_address end as call_trace_address_filled 
+            FROM {{ source('sudoswap_v2_ethereum','LSSVMPair_call_swapTokenForSpecificNFTs') }}
+            WHERE call_success
+            {% if is_incremental() %}
+            AND call_block_time >= date_trunc('day', now() - interval '7' day)
+            {% endif %}
+            ) sp_start
+        LEFT JOIN (
+            SELECT * FROM {{ source('sudoswap_v2_ethereum','ExponentialCurve_call_getBuyInfo') }}
+            UNION ALL 
+            SELECT * FROM {{ source('sudoswap_v2_ethereum','LinearCurve_call_getBuyInfo') }}
+            UNION ALL 
+            SELECT * FROM {{ source('sudoswap_v2_ethereum','XykCurve_call_getBuyInfo') }}
+            UNION ALL 
+            SELECT * FROM {{ source('sudoswap_v2_ethereum','GDACurve_call_getBuyInfo') }}
+        ) sp ON sp_start.call_tx_hash = sp.call_tx_hash
+            AND sp_start.call_block_number = sp.call_block_number
+            AND sp_start.call_trace_address_filled = slice(sp.call_trace_address,1,cardinality(sp_start.call_trace_address_filled))
+            {% if is_incremental() %}
+            AND sp.call_block_time >= date_trunc('day', now() - interval '7' day)
+            {% endif %}
+        LEFT JOIN {{ source('sudoswap_v2_ethereum','RoyaltyEngine_call_getRoyalty') }} roy 
+            ON roy.call_tx_hash = sp.call_tx_hash
+            AND roy.call_block_number = sp.call_block_number
+            AND sp_start.call_trace_address_filled = slice(roy.call_trace_address,1,cardinality(sp_start.call_trace_address_filled))
+            {% if is_incremental() %}
+            AND roy.call_block_time >= date_trunc('day', now() - interval '7' day)
+            {% endif %}
+        LEFT JOIN pools p ON p.pool_address = sp_start.contract_address
+    )
+    
+    , trades as (
+        SELECT
+            'ethereum' as blockchain
+            , 'sudoswap' as project
+            , 'v2' as project_version
+            , call_block_time as block_time
+            , call_block_number as block_number
+            , call_tx_hash as tx_hash
+            , pool_address as project_contract_address
+            , case when trade_category = 'sell' then pool_address else trade_recipient end as buyer
+            , case when trade_category = 'buy' then pool_address else trade_recipient end as seller
+            , nft_contract_address
+            , one_nft_token_id as nft_token_id
+            , case when nft_type = 'ERC721' then 1 else numItems end as nft_amount
+            , case when numItems > 0 then 'multiple' else 'single' end as trade_type
+            , trade_category
+            , token_contract_address as currency_contract
+            , cast(amount_raw/numItems as uint256) as price_raw
```

Comment:
for ERC721 this seems to be correct, but for ERC1155 we don't have to divide by the amount of tokens.

`price_raw` = amount payed for `nft_amount` tokens of ID `nft_token_id`
So for the example tx above:
- price_raw = 22.1 PRIME
- nft_amount = 10
- nft_token_id = 10318

0xa253119c7748456c7d383fcbdd03c0cf4897b5d59c697443932bfc18aab855a0

Reply:
price_raw here is not the spot price, but instead the total price because I'm taking the reverse percentage of platform fee. So to get the spot price average paid across the erc1155 tokens, you do have to divide by num items.

Reply:
please reread my comment, for erc1155 we don't need the spot price (price for 1 token) but the total price payed for the batch. 

This is needed so that `sum(price) = volume`

Reply:
guess it applies to all the fee calculations too, no need to divide